crates/context_aware_config/src/api/context/handlers.rs

```diff
@@ -463,17 +477,23 @@ async fn move_handler(
     req: Json<MoveReq>,
     mut db_conn: DbConnection,
     user: User,
-) -> superposition::Result<Json<PutResp>> {
+) -> superposition::Result<CustomResp<PutResp>> {
     let tags = parse_config_tags(custom_headers.config_tags)?;
     db_conn.transaction::<_, superposition::AppError, _>(|transaction_conn| {
         let move_reponse = r#move(path.into_inner(), req, transaction_conn, true, &user)
-            .map(Json)
             .map_err(|err| {
                 log::info!("move api failed with error: {:?}", err);
                 err
             })?;
-        add_config_version(&state, tags, transaction_conn)?;
-        Ok(move_reponse)
+        let version_id = add_config_version(&state, tags, transaction_conn)?;
+        let headers = HashMap::from([(
+            AppHeader::XConfigVersion.to_string(),
+            version_id.to_string(),
+        )]);
+        Ok(CustomResp {
+            body: move_reponse,
+            headers: headers,
+        })
```

Comment:
This whole part should be a function that takes the header enum and the response and wraps it appropriately

Reply:
@Datron we have to construct the headerMap inside handler only , then only thing left is to construct CustomResp, do we need another function for this ?